pkg/adapter/kubernetes/blueprintcr/blueprintSpecCRRepository.go

```diff
@@ -110,30 +109,59 @@ func (repo *blueprintSpecRepo) Update(ctx context.Context, spec *domain.Blueprin
 		return err
 	}
 
+	blueprintJson, err := repo.blueprintSerializer.Serialize(spec.Blueprint)
+	if err != nil {
+		return domainservice.NewInternalError(err, "failed to serialize blueprint json")
+	}
+
+	blueprintMaskJson, err := repo.blueprintMaskSerializer.Serialize(spec.BlueprintMask)
+	if err != nil {
+		return domainservice.NewInternalError(err, "failed to serialize blueprint mask json")
+	}
+
 	updatedBlueprint := v1.Blueprint{
-		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:              spec.Id,
 			ResourceVersion:   persistenceContext.resourceVersion,
 			CreationTimestamp: metav1.Time{},
 		},
-		Status: v1.BlueprintStatus{
-			Phase:              spec.Status,
-			EffectiveBlueprint: effectiveBlueprint,
-			StateDiff:          v1.ConvertToStateDiffDTO(spec.StateDiff),
+		Spec: v1.BlueprintSpec{
+			Blueprint:                blueprintJson,
+			BlueprintMask:            blueprintMaskJson,
+			IgnoreDoguHealth:         spec.Config.IgnoreDoguHealth,
+			IgnoreComponentHealth:    spec.Config.IgnoreComponentHealth,
+			AllowDoguNamespaceSwitch: spec.Config.AllowDoguNamespaceSwitch,
+			DryRun:                   spec.Config.DryRun,
 		},
 	}
 
 	logger.Info("update blueprint", "blueprint to save", updatedBlueprint)
-	CRAfterUpdate, err := repo.blueprintClient.UpdateStatus(ctx, &updatedBlueprint, metav1.UpdateOptions{})
+	CRAfterUpdate, err := repo.blueprintClient.Update(ctx, &updatedBlueprint, metav1.UpdateOptions{})
```

Comment:
A `retry.OnConflict` would make the method more robust.

Reply:
To ignore `ConflictErrors` with `retry.OnConflict` can lead to serious bugs. Therefore we decided to not do that in the blueprint-Operator.